src/PostsIterator.php

```diff
@@ -12,10 +12,10 @@
 class PostsIterator extends ArrayIterator
 {
     /**
-     * @var Post The last post that was returned by the iterator. Used
+     * @var null|Post The last post that was returned by the iterator. Used
      *                   to skip the logic in `current()`.
      */
-    protected Post $last_post;
+    protected $last_post;
```

Comment:
You still use a nullable type here.
```suggestion
    protected ?Post $last_post;
```